docs/src/main/asciidoc/_admin-guide/management-tasks/Configuration_file_git_history.adoc

```diff
@@ -124,3 +124,139 @@ You may need to push your repository changes to a remote repository so you can s
 [standalone@localhost:9990 /] :publish-configuration(location="origin")
 {"outcome" => "success"}
 ----
+
+[[ssh_auth]]
+== SSH Authentication
+
+In order to connect to any SSH git server to manage your configuration file history, you must use an Elytron configuration
+file to specify your SSH credentials. There are a number of ways to specify these credentials:
+
+=== SSH Key Location Credential
+
+It is possible to reference a file containing your SSH keys as follows:
+
+[source,xml,options="nowrap"]
+----
+<?xml version="1.0" encoding="UTF-8"?>
+<configuration>
+    <authentication-client xmlns="urn:elytron:client:1.4">
+        <authentication-rules>
+            <rule use-configuration="test-login">
+            </rule>
+        </authentication-rules>
+        <authentication-configurations>
+            <configuration name="test-login">
+                <credentials>
+                    <ssh-private-key-file ssh-directory="/home/user/git-persistence/" private-key-identity="id_ec_test">
+                        <clear-passphrase password="secret"/>
+                    </ssh-key-location>
+                </credentials>
+            </configuration>
+        </authentication-configurations>
+    </authentication-client>
+</configuration>
+----
+This configuration indicates that the private key to be used for SSH authentication is in the file `id_ec_test` in the
+directory `/home/user/git-persistence` and this private key was encrypted using the passphrase "secret".
+
+The `ssh-private-key-file`
+credential also take the following attributes:
+
+* `home-directory` - the home directory containing the `.ssh` directory. This is to be used if you would like to specify
+just the home directory to be used. `ssh-directory` is to be used to specify the complete path to the directory containing
+the SSH key. By default the home directory will be taken to be `~` and by default the SSH directory will be taken as `.ssh`
+under your home directory
+* `known-hosts-file` - this is to specify the name of the file containing the known SSH hosts you trust. This file should
+be in the same location as the private key file. By default, this will check `known_hosts` and `known_hosts2`
+* The default private key identities used are: `id_rsa`, `id_dsa`, `id_ecdsa`, and `id_ed22519`.
+
+One of the following elements may also be used to specify the passphrase used to encrypt the private key (if applicable):
```

Comment:
I think this should be reworded to indicate that this is the passphrase used to decrypt the private key if needed.

Reply:
Fixed, thanks!